---
title: SyntaxError
slug: orphaned/Web/JavaScript/Reference/Global_Objects/SyntaxError
tags:
  - Błąd składniowy
  - JavaScript
  - Klasa
  - Obiekt
  - Referencja
  - SyntaxError
translation_of: Web/JavaScript/Reference/Global_Objects/SyntaxError
original_slug: Web/JavaScript/Reference/Global_Objects/SyntaxError
---
<div>{{JSRef}}</div>

<p>Obiekt <code><strong>SyntaxError</strong></code> reprezentuje błąd przy próbie interpretacji kodu niepoprawnego pod względem składni. Jest wyrzucany wtedy, gdy silnik JavaScript podczas interpretacji kodu napotyka znak lub ciąg znaków niezgodny ze składnią języka.</p>

<h2 id="Konstruktor">Konstruktor</h2>

<dl>
 <dt><a href="/en-US/docs/Web/JavaScript/Reference/Global_Objects/SyntaxError/SyntaxError"><code>SyntaxError()</code></a></dt>
 <dd>Tworzy nowy obiekt <code>SyntaxError</code>.</dd>
</dl>

<h2 id="Właściwości_instancji">Właściwości instancji</h2>

<dl>
 <dt>{{jsxref("Error.prototype.message", "SyntaxError.prototype.message")}}</dt>
 <dd>Wiadomość błędu. Chociaż ECMA-262 mówi, że {{jsxref("SyntaxError")}} powinien dostarczać własną właściwość <code>message</code>, w <a href="/en-US/docs/Mozilla/Projects/SpiderMonkey">SpiderMonkey</a> jest ona dziedziczona z {{jsxref("Error.prototype.message")}}.</dd>
 <dt>{{jsxref("Error.prototype.name", "SyntaxError.prototype.name")}}</dt>
 <dd>Nazwa błędu. Odziedziczone z {{jsxref("Error")}}.</dd>
 <dt>{{jsxref("Error.prototype.fileName", "SyntaxError.prototype.fileName")}}</dt>
 <dd>Ścieżka do pliku zawierającego błąd. Odziedziczone z {{jsxref("Error")}}.</dd>
 <dt>{{jsxref("Error.prototype.lineNumber", "SyntaxError.prototype.lineNumber")}}</dt>
 <dd>Numer zawierającej błąd linii w pliku. Odziedziczone z {{jsxref("Error")}}.</dd>
 <dt>{{jsxref("Error.prototype.columnNumber", "SyntaxError.prototype.columnNumber")}}</dt>
 <dd>Numer zawierającej błąd kolumny w linii kodu. Odziedziczone z {{jsxref("Error")}}.</dd>
 <dt>{{jsxref("Error.prototype.stack", "SyntaxError.prototype.stack")}}</dt>
 <dd>Stos wywołań. Odziedziczone z {{jsxref("Error")}}.</dd>
</dl>

<h2 id="Przykłady">Przykłady</h2>

<h3 id="Obsługa_SyntaxError">Obsługa <code>SyntaxError</code></h3>

<pre class="brush: js">try {
  eval('hoo bar');
} catch (e) {
  console.error(e instanceof SyntaxError);
  console.error(e.message);
  console.error(e.name);
  console.error(e.fileName);
  console.error(e.lineNumber);
  console.error(e.columnNumber);
  console.error(e.stack);
}
</pre>

<h3 id="Tworzenie_SyntaxError">Tworzenie <code>SyntaxError</code></h3>

<pre class="brush: js">try {
  throw new SyntaxError('Hej', 'jakisPlik.js', 10);
} catch (e) {
  console.error(e instanceof SyntaxError); // true
  console.error(e.message);                // Hej
  console.error(e.name);                   // SyntaxError
  console.error(e.fileName);               // jakisPlik.js
  console.error(e.lineNumber);             // 10
  console.error(e.columnNumber);           // 0
  console.error(e.stack);                  // @debugger eval code:3:9
}
</pre>

<h2 id="Specyfikacje">Specyfikacje</h2>

<table class="standard-table">
 <thead>
  <tr>
   <th scope="col">Specyfikacja</th>
  </tr>
 </thead>
 <tbody>
  <tr>
   <td>{{SpecName('ESDraft', '#sec-native-error-types-used-in-this-standard-syntaxerror', 'SyntaxError')}}</td>
  </tr>
 </tbody>
</table>

<h2 id="Wsparcie_przeglądarek">Wsparcie przeglądarek</h2>

<div>


<p>{{Compat("javascript.builtins.SyntaxError")}}</p>
</div>

<h2 id="Zobacz_też">Zobacz też</h2>

<ul>
 <li>{{jsxref("Error")}}</li>
</ul>
